include/swift/SIL/SILConstants.h

```diff
@@ -197,6 +207,15 @@ class SymbolicValue {
     /// This is the number of bits in an RK_Integer or RK_IntegerInline
     /// representation, which makes the number of entries in the list derivable.
     unsigned integer_bitwidth;
+
+    /// This is the numer of bytes for an RK_String representation.
+    unsigned string_numBytes;
+
+    /// This is the number of elements for an RK_Aggregate representation.
+    unsigned aggregate_numElements;
+
+    /// This is the number of elements for an RK_Array representation.
+    unsigned array_numElements;
```

Comment:
a design level question: what if we unify `aggregate_numElements` and `array_numElements` into the same `numElements` field (perhaps even include `string_numBytes`)? it'd make the code a bit shorter -- would you say that'd be simpler as well?

Reply:
That's a defensible approach.  The reason I did it this way is that this is all a discriminated union where the aux and value fields have to line up.  This is easier to tell when the names directly line up like they are now.  There is no conceptual sharing of the values here, it is just that there happen to be a lot of things that want "numElements"